src/qibotn/backends/quimb.py

```diff
@@ -76,9 +104,17 @@ def execute_circuit(
                 NotImplementedError, "QiboTN quimb backend cannot support expectation"
             )
 
-        state = eval.dense_vector_tn_qu(
-            circuit.to_qasm(), initial_state, self.mps_opts, backend="numpy"
-        )
+        if TEBD_enabled:
+            nqubits = circuit.nqubits
+            if TEBD_option == True:
+                state = eval.tebd_tn_qu(circuit, self.tebd_opts)
```

Comment:
Being TEBD an evolution approximation, I'd hope that it should not need a dedicated evaluation. Possibly the same should be true for MPS, since in both cases the network generated will differ from the dense vector with just gates application, but once you got the network, you should evaluate the resulting observable(s) in a completely analogue way.

I'm not sure whether the information that you have a specific type of network may be used or not to improve the contraction. But, if yes, I'd propagate separately from the network generation (like maintaining an MPS/TEBD flag in the network representation, or something like that).